pkg/service/appvendor/resource_control.go

```diff
@@ -0,0 +1,139 @@
+package appvendor
+
+import (
+	"context"
+	"time"
+
+	"openpitrix.io/openpitrix/pkg/constants"
+	"openpitrix.io/openpitrix/pkg/db"
+	"openpitrix.io/openpitrix/pkg/manager"
+	"openpitrix.io/openpitrix/pkg/models"
+	"openpitrix.io/openpitrix/pkg/pb"
+	"openpitrix.io/openpitrix/pkg/pi"
+	"openpitrix.io/openpitrix/pkg/util/pbutil"
+)
+
+type Resource struct {
+}
+
+func (resource *Resource) DescribeVendorVerifyInfos(ctx context.Context, req *pb.DescribeVendorVerifyInfosRequest) ([]*models.Vendor, uint32, error) {
+	var vendors []*models.Vendor
+	offset := pbutil.GetOffsetFromRequest(req)
+	limit := pbutil.GetLimitFromRequest(req)
+
+	var vendorColumns = db.GetColumnsFromStruct(&models.Vendor{})
+	query := pi.Global().DB(ctx).
+		Select(vendorColumns...).
+		From(constants.TableVendor).
+		Offset(offset).
+		Limit(limit).
+		Where(manager.BuildFilterConditions(req, constants.TableVendor))
+
+	query = manager.AddQueryOrderDir(query, req, constants.ColumnCreateTime)
+	_, err := query.Load(&vendors)
+	count, err := query.Count()
+	return vendors, count, err
+}
+
+func (resource *Resource) GetVendorVerifyInfo(ctx context.Context, userID string) (*models.Vendor, error) {
+	vendor := &models.Vendor{}
+	vendorColumns := db.GetColumnsFromStruct(&models.Vendor{})
+	err := pi.Global().DB(ctx).
+		Select(vendorColumns...).
+		From(constants.TableVendor).
+		Where(db.Eq(constants.ColumnUserId, userID)).
+		LoadOne(&vendor)
+	if err != nil {
+		return nil, err
+	}
+	return vendor, nil
+}
+
+func (resource *Resource) SubmitVendorVerifyInfo(ctx context.Context, vendor models.Vendor) (string, error) {
+	var modifyType string
+	var err error
+	var userID string
+
+	info, err := resource.GetVendorVerifyInfo(ctx, vendor.UserId)
+	if info == nil {
+		modifyType = "new"
+	} else {
+		modifyType = "update"
+	}
+
+	if modifyType == "new" {
+		userID, err = resource.CreateVendorVerifyInfo(ctx, vendor)
+		return userID, err
+	}
+	if modifyType == "update" {
+		userID, err = resource.UpdateVendorVerifyInfo(ctx, vendor)
+		return userID, err
+	}
+	return "", err
+}
+
+func (resource *Resource) UpdateVendorVerifyInfo(ctx context.Context, vendor models.Vendor) (string, error) {
```

Comment:
It is not necessary to update all fields, it is better to update inconsistent fields.